omegaconf/omegaconf.py

```diff
@@ -101,43 +100,67 @@ def SI(interpolation: str) -> Any:
 
 
 def register_default_resolvers() -> None:
-    def env(key: str, default: Any = _EMPTY_MARKER_) -> Any:
+    # DEPRECATED: remove in 2.2
+    def legacy_env(key: str, default: Optional[str] = None) -> Any:
+        warnings.warn(
+            "The `env` resolver is deprecated, see https://github.com/omry/omegaconf/issues/573",
+        )
+
         try:
-            val_str = os.environ[key]
+            return decode_primitive(os.environ[key])
         except KeyError:
-            if default is not _EMPTY_MARKER_:
-                return default
+            if default is not None:
+                # We handle the "null" string in a special way so as to preserve
+                # the old behavior of `env` while keeping the fix from #230.
+                if default.lower() == "null":
+                    return None
+                else:
+                    return decode_primitive(default)
```

Comment:
OmegaConf 2.0 does not treat null in a special way:
```python

In [1]: from omegaconf import *

In [2]: __version__
Out[2]: '2.0.6'

In [3]: OmegaConf.create({"a": "${env:ABC,null}"}).a
Out[3]: 'null'
```

#230 is for 2.1, there is no need for the legacy resolver to change behavior.

Reply:
In the meantime, I removed this feature in 459a734

Reply:
This actually also requires updating news items: done in e6342df